### ISSUE TITLE
Attribute values starting with 'service' (and others) create broken configuration

### ISSUE THREAD

**Issue by tinuzz** [voxpupuli/puppet-icinga2#614]
I am trying to configure a HTTP check for a vhost named 'service.example.com', and it produces a broken configuration.

```
    ::icinga2::object::service { "test_http_service.example.com":
        target           => '/etc/icinga2/conf.d/services/test_services.conf',
        host_name        => 'eridium.aws',
        display_name     => 'HTTP service.example.com',
        import           => [ 'generic-service' ],
        check_command    => 'http',
        vars             => {
            'http_vhost' => 'service.example.com',
            'http_ssl'   => true,
            'http_sni'   => true,
        },
    }
```
This results in the following broken configuration:

```
object Service "test_http_service.example.com" {
  import "generic-service"

  display_name = "HTTP service.example.com"
  host_name = "eridium.aws"
  check_command = "http"
  vars.http_vhost = service.example.com
  vars.http_ssl = true
  vars.http_sni = true
}
```

## Expected Behavior

The value of the resulting 'vars.http_vhost' should have quotes surrounding it:

```
  vars.http_vhost = "service.example.com"
```

## Current Behavior

The value is missing quotes.

## Possible Solution

I think this is a bug in the attibute value parser in [lib/puppet_x/icinga2/utils.rb, line 171](https://github.com/Icinga/puppet-icinga2/blob/master/lib/puppet_x/icinga2/utils.rb#L171):

```
if value =~ /^-?\d+\.?\d*[dhms]?$/ || value =~ /^(true|false|null)$/ || value =~ /^!?(host|service|user)\./ || value =~ /^\{{2}.*\}{2}$/
```

Certain values, including all values that start with 'service.' are treated special and returned without quotes.

I am not sure what use case this is meant to support (just starting out using Icinga2 and this module), but it seems to me that the check should be stricter, or there should be a way to force the value as a simple string (is there?).

In my testing, I also noted that using brackets in a value also produces unexpected results:

```
display_name     => 'HTTP service.example.com (test)'
```

turns into:

```
display_name = HTTP service.exampe.com ("test")
```

which has no surrounding quotes but quotes around the part of the value between brackets.

## Your Environment

* Module version (`puppet module list`): 2.4.0
* Puppet version (`puppet -V`): 6.14.0
* Operating System and version: Debian 10 (Buster)

**Comment by lbetz** [comment]
Thx for your report. You have identified the correct line but that's not a bug. The reason is described in the documentation under https://github.com/Icinga/puppet-icinga2#how-configuration-is-parsed. Your solution is to use -: to disable the parser.

display_name     => '-:"HTTP service.example.com (test)"'
...
  vars             => {
            'http_vhost' => '-:"service.example.com"',
...

**Comment by tinuzz** [comment]
Oops, I'm sorry, I didn't read that. Didn't expect anything like that to be the default behaviour though, that's why I went bughunting in the source. Thanks.

**Comment by tinuzz** [comment]
Hi, sorry to add another reply after closing the issue and I'm not even sure it's going to be helpful, but I just wanted to say, after trying to work with (around) this parser, that it strikes me as extremely odd to make this thing opt-out, rather than opt-in.

Since lots of data that is passed into these parameters isn't necessarily trusted not to contain any special characters, I don't really have a choice but to disable the parser almost everywhere. I now have Puppet code littered with uglyness like this:

```display_name  => "-:\"$display_name\"",```

because even the quotes are gone after disabling the parser.

Or even worse, because this requirement now turns _undef_ values into empty strings, or strings containing only '-:', I now have to do this:

```'http_onredirect' => $onredirect ? { undef => undef, default => "-:\"$onredirect\"" },```

Is this really necessary, or am I missing something here?

Edit: I removed a piece about an observed difference between 'object' and 'service', but that was due to a mistake on my part. I double-escaped the value (added '-:' at two levels), leaving one '-:' in place. This is a reenforcement of my statement that having to disable the parser on a per-attribute level is extremely cumbersome and error-prone.
